src/host/outputStream.cpp

```diff
@@ -221,27 +221,55 @@ void ConhostInternalGetSet::ShowWindow(bool showOrHide)
 }
 
 // Routine Description:
-// - Connects the SetConsoleOutputCP API call directly into our Driver Message servicing call inside Conhost.exe
+// - Set the code page used for translating text when calling A versions of I/O functions.
 // Arguments:
-// - codepage - the new output codepage of the console.
+// - codepage - the new code page of the console.
 // Return Value:
 // - <none>
-void ConhostInternalGetSet::SetConsoleOutputCP(const unsigned int codepage)
+void ConhostInternalGetSet::SetCodePage(const unsigned int codepage)
 {
-    THROW_IF_FAILED(DoSrvSetConsoleOutputCodePage(codepage));
+    if (IsValidCodePage(codepage))
+    {

```

Comment:
Is there a reason why we should check this outside the two set functions? I think it may be safer if it was inside them.

Reply:
It just seemed a little more efficient, but that's not really important here. I've moved the checks back into the `DoSrvSet*` functions now.